### PR TITLE
Q3 2022 repository standardization

### PR DESCRIPTION
**Pull request by rekmarks**
Standardizes this repository per the module template as of Q3 2022. All changes are copied over directly from the template, except for the tests, which have been rewritten. A couple of notes:
- I didn't bother rewriting the tests for JSON-RPC notifications, as that functionality will be implemented in a completely different manner in #24 has been merged.
- CI has been switched from CircleCI to GitHub Actions.

### CODE REVIEW

**Comment by rekmarks** [comment]
Status checks are failing because of the switch to GitHub Actions. I'll fix this once I have an approval.

File: tsconfig.json
Comment (mcmire) [inline tsconfig.json:12]: We shouldn't need this setting, as this is the default value: https://www.typescriptlang.org/tsconfig#typeRoots

**Review by mcmire — COMMENTED** [review]
I wasn't aware we had this package 😅 One small thing but otherwise looks good to me.

File: tsconfig.json
Comment (mcmire) [reply tsconfig.json:12]: Well, take that back, this tells TypeScript to use `@types/*` packages that are located immediately within `node_modules` as opposed to anywhere else, but we don't usually have to mess with this setting.

Comment (rekmarks) [reply tsconfig.json]: Yeah, good point. I just preserved it because it was already there. Removed in [042dd7a](https://github.com/MetaMask/json-rpc-middleware-stream/pull/25/commits/042dd7a4ef1e32ccd5ee437b50d90330062b877a)

**Review by mcmire — APPROVED** [review]
This looks good to me! Note that there are two more PRs I merged into the module template just now — one to upgrade to Yarn v3, another to tweak a TypeScript setting. Up to you on whether you want to include those now or later. As for this PR though, looks good.

**Pull request merged**
